File: dailyChallenges/0017-HappyNumber/17-HappyNumber.go
func isHappy(n int) bool {
	slow, fast := n, squareOfNum(n)

	for fast != 1 && slow != fast {
		slow = squareOfNum(slow)
		fast = squareOfNum(squareOfNum(fast))
	}
	return fast == 1
}

func squareOfNum(n int) int {
	output := 0

	for n > 0 {
		digit := n % 10
		output += digit * digit
		n /= 10
	}
	return output
}

File: dailyChallenges/0017-HappyNumber/17-HappyNumber.cpp
class Solution
{
public:
    bool isHappy(int n) 
    {
        int slow = n;
        int fast = squareOfNum(n);

        while(slow!=fast && fast != 1)
        {
            slow = squareOfNum(slow);
            fast = squareOfNum(squareOfNum(fast));
        }
        if(fast == 1)
        {
            return true;
        }
        return false;
    }
private:
    int squareOfNum(int n)
    {
        int output = 0;
        while(n>0)
        {
            int digit = n % 10;
            digit = pow(digit, 2);
            output += digit;
            n /= 10;
        }
        return output;
    }
};